### PR TITLE
Support for MQTT-C

### PR DESCRIPTION
**Pull request by Ouss4**
## Summary
Add glue code for MQTT-C and an example to use the library.
## Impact
No impact on existing code.  Library disabled by default.
## Testing
Tested with ESP32 DevkitC and Olimex STM32 P407. defconfigs attached, IP addresses need to be adapted.
[esp32_mqttc.txt](https://github.com/apache/incubator-nuttx-apps/files/6220719/esp32_mqttc.txt)
[olimex_mqttc.txt](https://github.com/apache/incubator-nuttx-apps/files/6220720/olimex_mqttc.txt)

### CODE REVIEW

File: mqtt/mqttc/Make.defs
Comment (Ouss4) [inline mqtt/mqttc/Make.defs:25]: Should we contribute `__NuttX__` to the upstream library instead?  We've got similar situations with other 3rd party software as well. [mbedTLS](https://github.com/apache/incubator-nuttx-apps/pull/566/files#diff-6043c39303f7b14be00e75ffcac170a85c3ee0b7369e7db85f4131ef85280c81R39) is one of them.

**Comment by Ouss4** [comment]
Mixed Case style error comes from MQTT-C APIs.

File: mqtt/mqttc/Make.defs
Comment (v01d) [reply mqtt/mqttc/Make.defs:25]: Yeah, I think it is a simple change to request. In the meantime we can leave this definition.

**Comment by v01d** [comment]
Thanks for submitting. I will test it on sim if I can (otherwise on ESP32).
BTW, we discussed putting the library under netutils on the list, could you move it there so that we not crowd to toplevel directory?
I think this will make the include path change to `netutils/mqtt.h`, unless we leave it in top-level include for compatibility.

File: mqtt/mqttc/Make.defs
Comment (xiaoxiang781216) [reply mqtt/mqttc/Make.defs:25]: ```__unix__``` is only used in implementation, it's better to move CLFAGS/CXXFLAGS into Makefile

File: mqtt/mqttc/Kconfig
Comment (xiaoxiang781216) [inline mqtt/mqttc/Kconfig:6]: should we add NETUTILS_ prefix?

Comment (xiaoxiang781216) [reply mqtt/mqttc/Kconfig:6]: change MQTT_MQTTC to MQTTC?

File: mqtt/mqttc/Make.defs
Comment (xiaoxiang781216) [inline mqtt/mqttc/Make.defs:2]: change path to netutils/mqttc?

File: examples/mqttc/Kconfig
Comment (xiaoxiang781216) [inline examples/mqttc/Kconfig:21]: Can we use the default stack size(2KB) here?

**Review by xiaoxiang781216 — APPROVED** [review]

File: mqtt/mqttc/Make.defs
Comment (xiaoxiang781216) [reply mqtt/mqttc/Make.defs]: @Ouss4 how about this comment?

File: examples/mqttc/Kconfig
Comment (Ouss4) [reply examples/mqttc/Kconfig:21]: I'm not sure if that's enough.  I remember that I had to adjust the stack size a while back when I had this example.  I'll do some tests later today and see if 2KB is enough.

File: mqtt/mqttc/Make.defs
Comment (Ouss4) [reply mqtt/mqttc/Make.defs]: @xiaoxiang781216 tried to move them (as is done in other places) but they don't seem to take any effect in Makefile.  Any thing to look for here?

Comment (xiaoxiang781216) [reply mqtt/mqttc/Make.defs]: I just create a patch to mainline: https://github.com/LiamBindle/MQTT-C/pull/131
We can remove these lines once the patch get merged, let's ignore it now.

Comment (Ouss4) [reply mqtt/mqttc/Make.defs]: But we'll still need to wait for the next release.  Or use master?

**Comment by Ouss4** [comment]
I updated the defconfigs in the description to account for the new location.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**
